Insurance pools of closed / settled markets to go to on - chain treasury (#2065)
vegaprotocol/specs: protocol/0043-MKTL-market_lifecycle.md

### Cancelled

A market becomes Cancelled when a Market Proposal is successful and conditions are not met to transition the Market to the Active state during the Pending period, and the trading terminated data source input is triggered, see [data sourcing](./0045-DSRC-data_sourcing.md).

> When a market transitions to a cancelled state all orders should be cancelled and collateral returned to respective parties general account for the relevant asset, all LP commitments should be cancelled and their bond returned to the general account for the relevant asset. After `market.liquidity.successorLaunchWindowLength` has elapsed since cancellation any insurance pool balance should get [redistributed](./0015-INSR-market_insurance_pool_collateral.md) into the global insurance pool using the same settlement asset.

Reviewer: ```suggestion
When a market transitions to a cancelled state all orders should be cancelled and collateral returned to respective parties general account for the relevant asset, all LP commitments should be cancelled and their bond returned to the general account for the relevant asset. After `market.liquidity.successorLaunchWindowLength` has elapsed since cancellation any [insurance pool](./0015-INSR-market_insurance_pool_collateral.md) balance should get transferred into the global insurance pool using the same settlement asset.
```

[LP fees](0042-LIQF-setting_fees_and_rewarding_lps.md) that have been cumulated but not yet paid out are distributed to the market LPs as per the LP spec.
After `market.liquidity.successorLaunchWindowLength` has elapsed since the settlement time

> - [Insurance pool funds](./0015-INSR-market_insurance_pool_collateral.md) are redistributed into the global insurance pool using the same settlement asset. For markets that have a named successor market the insurance pool balance is transferred to the insurance pool of the successor market.

Reviewer: ```suggestion
- [Insurance pool funds](./0015-INSR-market_insurance_pool_collateral.md) are transferred (after window for successor creation elapsed) into the global insurance pool using the same settlement asset. For markets that have a named successor market the appropriate fraction of the insurance pool balance is transferred to the insurance pool of the successor market.
```

Parties that had open positions see settlement cash-flows happen.
Margin account balances are transferred to the general account.
The market state is `settled`.

> After `market.liquidity.successorLaunchWindowLength` has passed since market settlement, any insurance pool balance is [redistributed](./0015-INSR-market_insurance_pool_collateral.md) into the global insurance pool using the same settlement asset.

Reviewer: ```suggestion
After `market.liquidity.successorLaunchWindowLength` has passed since market settlement, any [insurance pool](./0015-INSR-market_insurance_pool_collateral.md) balance is transferred into the global insurance pool using the same settlement asset.
```

When this is approved and enacted the market state is `closed`.
Parties that had open positions see settlement cash-flows happen to settle positions.
Margin account balances are transferred to the general account.

> After `market.liquidity.successorLaunchWindowLength` has passed since market settlement, any insurance pool balance is [redistributed](./0015-INSR-market_insurance_pool_collateral.md) into the global insurance pool using the same settlement asset.

Reviewer: ```suggestion
After `market.liquidity.successorLaunchWindowLength` has passed since market settlement, any [insurance pool](./0015-INSR-market_insurance_pool_collateral.md) balance is transferred into the global insurance pool using the same settlement asset.
```

### Market never leaves opening auction, trading terminated triggered, market cancelled (<a name="0043-MKTL-010" href="#0043-MKTL-010">0043-MKTL-010</a>)

1. A market is proposed, approved by governance process and enters the opening auction (Pending state).
1. Trading terminated data source triggers before the market leaves the opening auction (so market never left Pending state so far).
1. All orders should be cancelled and collateral returned to respective parties general account for the relevant asset.
1. All LP commitments should be cancelled and their bond returned to the general account for the relevant asset.

> 1. After `market.liquidity.successorLaunchWindowLength` has elapsed since market cancellation, any insurance pool balance should be [redistributed](./0015-INSR-market_insurance_pool_collateral.md) into the global insurance pool using the same settlement asset.

Reviewer: ```suggestion
1. After `market.liquidity.successorLaunchWindowLength` has elapsed since market cancellation, any insurance pool balance should be transferred into the global insurance pool using the same settlement asset.
```
